### PR TITLE
stm32_common/flashpage: cleanup

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

This PR cleans up `flashpage.c`. Following discussion in https://github.com/RIOT-OS/RIOT/pull/11681 it changes defines to makes specific behaviors explicit. Some unused functions and definitions where removed, specifically for stm32l1/0 and stm32l4.

NOTE: to verify the stm32l1/l0 change you can refer to the RM p71 for L1 and p89 for L0x3 and p81 for L0x1.

### Testing procedure

Run `make -C tests/periph_flashpage/ BOARD=nucleo-l476rg flash test` for an `stm32l1`, `stm32l0` and `stm32l4` , `stm32f0` and `stm32f1` board.

The test should still pass in all cases.

Tested on: `nucleo-l152re`, `b-l072rz-lrwan1`, `nucleo--l476rg`, `nucleo-f072rb`, `iotlab-m3`.

### CODE REVIEW

File: cpu/stm32_common/periph/flashpage.c [hunk 1]
```diff
@@ -77,7 +76,8 @@ static void _erase_page(void *page_addr)
     uint32_t *dst = page_addr;
 #else
     uint16_t *dst = page_addr;
-
+#endif
+#if defined(CPU_FAM_STM32F0) || defined(CPU_FAM_STM32F1)
```
Comment (aabadie) [inline cpu/stm32_common/periph/flashpage.c]: This change should be in it's own commit and not in b72a6edd87b6a438694beb43178c4f04347f824a where it's unrelated.

File: cpu/stm32_common/periph/flashpage.c [hunk 2]
```diff
@@ -185,14 +186,16 @@ void flashpage_write_raw(void *target_addr, const void *data, size_t len)
     }
 
     /* clear program bit again */
+#if defined(CPU_FAM_STM32F0) || defined(CPU_FAM_STM32F1) || \
+    defined(CPU_FAM_STM32L4)
     CNTRL_REG &= ~(FLASH_CR_PG);
     DEBUG("[flashpage_raw] write: done writing data\n");
+#endif
```
Comment (aabadie) [inline cpu/stm32_common/periph/flashpage.c]: The DEBUG message above is now only displayed with F0/F1/L4 but not for L0/L1. It should be put after.
I would also not put this change in 5e0c86720013de864829e3e0edc10ace3f058e0d because it seems unrelated.

Comment (fjmolinas) [reply cpu/stm32_common/periph/flashpage.c]: You are right about the `DEBUG` message. But it is related since I'm removing the definition of `FLASH_CR_PG` Therefore I need to excluded it from compilation and that is the reason for the `#ifdefined`. Should I split guarding and removing `FLASH_CR_PG` into two commits?

File: cpu/stm32_common/periph/flashpage.c (hunk 1 above)
Comment (fjmolinas) [reply cpu/stm32_common/periph/flashpage.c]: Why? Only `STM32F0 STM32F1` need to initialize HSI clock (at least for the implemented boards). My goal was to separate it from the default case which only said   ` /* Default is to support half-word sizes */`, But I guess you are right, the commit message is not explicit enough. Would you be Ok with just re-wording?

File: cpu/stm32_common/periph/flashpage.c (hunk 2 above)
Comment (aabadie) [reply cpu/stm32_common/periph/flashpage.c]: I see, let's keep the changes in the same commit then. You still need have to fix the commit message: it's full of typos :)

**Comment by aabadie** [comment]
There is also a typo in 9416ca8705ca050960864d2d5b96f8a83c84864d message

File: cpu/stm32_common/periph/flashpage.c (hunk 1 above)
Comment (aabadie) [reply cpu/stm32_common/periph/flashpage.c]: No, this changes the initial behaviour as you say, so it needs to be in its own commit.

**Comment by aabadie** [comment]
I tested this on nucleo-l432kc, nucleo-l073rz, nucleo-l433rc and iotlab-m3. It worked.

Just rework a bit the commits and we are good to go.

**Comment by fjmolinas** [comment]
@aabadie Changes addressed, I split the commit you where addressing into two. One separating the hsi handling case from the default case. And the second one making the remaining implicit defines explicit, what do you think?

File: cpu/stm32_common/periph/flashpage.c (hunk 2 above)
Comment (aabadie) [reply cpu/stm32_common/periph/flashpage.c]: The DEBUG line must still move out of the define (preferably after).

Comment (fjmolinas) [reply cpu/stm32_common/periph/flashpage.c]: damn, I was concentrating on the typos sorry

Comment (fjmolinas) [reply cpu/stm32_common/periph/flashpage.c]: Ok I think now it should be ok .. sorry for missing this.
